src/signals/incident/components/IncidentForm/index.js

```diff
@@ -52,6 +52,8 @@ class IncidentForm extends Component {
       if (this.form.valid) {
         this.setIncident(this.state.formAction)
         this.state.next()
+      } else {
+        this.handleSubmit(null, this.state.next, this.state.formAction)
```

Comment:
What is the use of this else statement? (and the if (e) in the handleSubmit ?

Reply:
This makes sure the form is validated and the results are shown on the page. Otherwise, there was a situation when logged in where you'd click the button, you stay on the page, but don't see any feedback. In this case, there's no event, hence the `if (e)` in the `handleSubmit`.

Reply:
I'm not convinced this is necessary, I've tried several things but never get into the else statement.
And it makes the code quite unreadable with the nested ifs.

Reply:
You are right. The else was indeed not necessary!